posthog/temporal/data_imports/sources/linkedin_ads/service/service.py

```diff
@@ -0,0 +1,318 @@
+        }
+
+    def _prepare_analytics_dates(
+        self,
+        should_use_incremental_field: bool,
+        incremental_field: Optional[str],
+        incremental_field_type: Optional[IncrementalFieldType],
+        db_incremental_field_last_value: Any,
+        date_start: Optional[str],
+        date_end: Optional[str],
+        sync_frequency_interval: Optional[dt.timedelta],
+    ) -> tuple[Optional[str], Optional[str]]:
+        """Prepare date range for analytics requests.
+
+        Args:
+            should_use_incremental_field: Whether to use incremental sync
+            incremental_field: Field name for incremental sync
+            incremental_field_type: Field type for incremental sync
+            db_incremental_field_last_value: Last value for incremental sync
+            date_start: Start date for analytics data
+            date_end: End date for analytics data
+            sync_frequency_interval: Sync frequency interval
+
+        Returns:
+            Tuple of (date_start, date_end) strings
+        """
+        if should_use_incremental_field and incremental_field and incremental_field_type:
+            # Determine last value using incremental_field_type
+            if db_incremental_field_last_value is None:
+                last_value = (dt.datetime.now() - dt.timedelta(days=30)).strftime("%Y-%m-%d")
+            else:
+                last_value = db_incremental_field_last_value
+
+            # For analytics (date-based), use incremental value as start date
+            if (
+                incremental_field_type == IncrementalFieldType.Date
+                and incremental_field == "dateRange.start"
+                and not date_start
+            ):
+                date_start = self.date_handler.calculate_incremental_date_range(last_value, sync_frequency_interval)
+
+        return date_start, date_end
+
+    def _create_source_response(self, resource_name: str, data: list[dict[str, Any]]) -> SourceResponse:
+        """Create SourceResponse object from fetched data.
+
+        Args:
+            resource_name: Name of the resource
+            data: Raw data from the API
+
+        Returns:
+            SourceResponse object for the pipeline
+        """
+        # Flatten the data structure
+        flattened_data = [flatten_data_item(item, resource_name) for item in data]
+
+        # Determine primary keys based on resource type
+        primary_keys = determine_primary_keys(resource_name, flattened_data)
+
+        return SourceResponse(
+            name=resource_name,
+            items=flattened_data,
+            primary_keys=primary_keys,
+            column_hints=None,
+            partition_count=1,
+            partition_size=1,
+            partition_mode="datetime",
+            partition_format="month",
+            partition_keys=["date_range_start"] if flattened_data and "date_range_start" in flattened_data[0] else None,
```

Comment:
Which endpoints don't have `date_range_start`? Is `date_range_start` always stable (e.g. it doesn't change)? For the endpoints that dont have `date_range_start`, what do they have instead?